transcom/mymove
# INT I-13494 B-21592 issue fix

## [Issue ticket](https://www13.v1host.com/USTRANSCOM38/Issue.mvc/Summary?oidToken=Issue%3A1029382)
## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21592)

## Summary

The man, myth, and legend @CoryKleinjanCACI found an issue following the initial merge of B-21592 that they were unable to approve full PPM shipments due to the lack of a `requestedPickupDate`. Since we want to ensure that the right base year is picked, I sent out a question to Dan Schuster and he confirmed that he wants to use a PPM's expected departure date value to decide the base year.

This PR does the following:
- updates query to use `Eager` when initially querying the move when creating a service item (`FetchOne` just wasn't cutting it since I couldn't use associations)
- added in 
```
	var nilTime time.Time
	if shipment.PPMShipment != nil && shipment.PPMShipment.ExpectedDepartureDate != nilTime {
		feeDate = &shipment.PPMShipment.ExpectedDepartureDate
	}
```
when creating a service item to cover down on PPM shipments
- added full PPM test

### How to test

1. Access MM as office/customer a create a move with full PPM (setting `NS Mayport` as current/destination)
2. Go in as the TOO, update/add orders information
3. Approve the PPM shipment and add the CS service item
4. Should be successful and should have an estimated price in the correct base year
5. Repeat with partial PPM
6. Should pull from NOT the PPM shipment (requested pickup date) for the correct base year


![Screenshot 2024-11-07 at 10 39 13 AM](https://github.com/user-attachments/assets/5a1fa7d6-4451-4bb0-bd4f-d9d90222cb0d)
